agent/pkg/bcloudapi/client.go

```diff
@@ -114,7 +114,7 @@ func (c *client) RegisterAgent(ctx context.Context, agentName string) (*AgentInf
 	defer rsp.Body.Close()
 
 	if rsp.StatusCode != http.StatusOK {
-		if apiErr := extractErrorFromResponse(rsp.Request.Response); apiErr != nil {
+		if apiErr := extractErrorFromResponse(rsp); apiErr != nil {
```

Comment:
@zaharidichev Do you happen to recall why we were trying to read from `rsp.Request.Response`? Are there certain failure modes where this struct is populated? If so, I might rework this to check both `rsp` and `rsp.Request.Response`.

For context, here was the observed crash:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x1665758]

goroutine 1 [running]:
github.com/buoyantio/linkerd-buoyant/agent/pkg/bcloudapi.extractErrorFromResponse(0xc00056a8d0)
	/bcloud-build/agent/pkg/bcloudapi/client.go:198 +0x18
github.com/buoyantio/linkerd-buoyant/agent/pkg/bcloudapi.(*client).RegisterAgent(0xc000130300, {0x1d1ae08, 0xc000058020}, {0xc000058798, 0x4})
	/bcloud-build/agent/pkg/bcloudapi/client.go:117 +0x2c8
github.com/buoyantio/linkerd-buoyant/agent/pkg/registrator.(*Registrator).EnsureRegistered(0xc000717ee0, {0x1d1ae08, 0xc000058020})
	/bcloud-build/agent/pkg/registrator/registrator.go:67 +0x318
github.com/buoyantio/linkerd-buoyant/agent/cmd/registrator.Main({0xc00004c0a0, 0x2, 0x2})
	/bcloud-build/agent/cmd/registrator/main.go:51 +0x3a7
main.main()
	/bcloud-build/agent/main.go:24 +0x116
```

Reply:
I cannot recall. I think this was just me not understanding the API... I do not think we should read from `Request.Response`. Thanks for fixing that.